GithubProfileWiki/Networking/APIService.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  APIService.swift
+//  GithubProfileWiki
+//
+//  Created by ilter on 23.01.2022.
+//
+
+import Foundation
+
+class APIService<T: APIHandler> {
+    let apiRequest: T
+    let urlSession: URLSession
+    
+    init(apiRequest: T, urlSession: URLSession = .shared) {
+        self.apiRequest = apiRequest
+        self.urlSession = urlSession
+    }
+    
+    func submitRequest(requestData: T.RequestDataType, completionHandler: @escaping (T.ResponseDataType?, Error?) -> ()) {
+        if let urlRequest = apiRequest.submitRequest(from: requestData) {
+            urlSession.dataTask(with: urlRequest) { ( data, response, error) in
+                guard let data = data,
+                      let httpResponse = response as? HTTPURLResponse else {
+                          return completionHandler(nil, error)
```

Comment:
an error handler can be implemented instead of carrying all the responsiblity to completion handler. For example yu have a generic error case but should you define it wherever you made a call?

Reply:
Yeah definitely I should add error handler 

- [ ]  Add generic error handler